test/unit/component/parser/json/BpmnJsonParser.event-utils.ts

```diff
@@ -0,0 +1,238 @@
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type { ExpectedBoundaryEventShape, ExpectedEventShape, ExpectedShape } from '../../../helpers/bpmn-model-expect';
+import type { BuildEventsParameter } from '../../../helpers/JsonBuilder';
+
+import { verifyShape } from '../../../helpers/bpmn-model-expect';
+import { buildDefinitions, EventDefinitionOn } from '../../../helpers/JsonBuilder';
+import {
+  expectAsWarning,
+  parseJsonAndExpectEvent,
+  parseJsonAndExpectOnlyEdgesAndFlowNodes,
+  parseJsonAndExpectOnlyFlowNodes,
+  parsingMessageCollector,
+} from '../../../helpers/JsonTestUtils';
+import { getEventShapes } from '../../../helpers/TestUtils';
+
+import { ShapeUnknownBpmnElementWarning } from '@lib/component/parser/json/warnings';
+import { ShapeBpmnEventDefinitionKind } from '@lib/model/bpmn/internal';
+
+/*
+  eslint-disable jest/no-export
+*/
+
+export type OmitExpectedEventShape = Omit<ExpectedEventShape, 'shapeId' | 'bpmnElementId' | 'bounds'> | Omit<ExpectedBoundaryEventShape, 'shapeId' | 'bpmnElementId' | 'bounds'>;
+
+export const expectedBounds = { x: 362, y: 232, width: 36, height: 45 };
+
+export function testMustConvertShapes(buildEventParameter: BuildEventsParameter | BuildEventsParameter[], omitExpectedShape: OmitExpectedEventShape, processIsArray = false): void {
+  const process = { event: buildEventParameter, task: {} };
+  const json = buildDefinitions({ process: processIsArray ? [process] : process });
+
+  const model = parseJsonAndExpectEvent(json, omitExpectedShape.eventDefinitionKind, Array.isArray(buildEventParameter) ? buildEventParameter.length : 1);
+
+  const shapes = getEventShapes(model);
+  for (const [index, shape] of shapes.entries()) {
+    const expectedShape: ExpectedEventShape = {
+      ...omitExpectedShape,
+      shapeId: `shape_event_id_0_${index}`,
+      bpmnElementId: `event_id_0_${index}`,
+      bounds: expectedBounds,
+    };
+    verifyShape(shape, expectedShape);
+  }
+}
+
+export function testMustNotConvertEvent(buildEventParameter: BuildEventsParameter): void {
+  const json = buildDefinitions({ process: { event: buildEventParameter, task: {} } });
+
+  const bpmnModel = parseJsonAndExpectOnlyFlowNodes(json, 1, 1);
+
+  expect(getEventShapes(bpmnModel)).toHaveLength(0);
+
+  const warning = expectAsWarning<ShapeUnknownBpmnElementWarning>(parsingMessageCollector.getWarnings()[0], ShapeUnknownBpmnElementWarning);
+  expect(warning.bpmnElementId).toBe('event_id_0_0');
+}
+
+export function executeEventCommonTests(buildEventParameter: BuildEventsParameter, omitExpectedShape: OmitExpectedEventShape, titleSuffix: string): void {
```

Comment:
**thought:** I am not sure that extracting this function will make things easier to understand in the actual test files (new indirection in an external file).
But, as we must split the tests into several files, I guess there is no better solution for now.

Reply:
Indeed, we need to switch files to read the tests. But, before, there is a lot of lines to understand what is the running tests and what is the utils.  
  
The other solution to do in another PR is to not have parametered tests like it is and duplicate them as for other BPMN elements 😊